docs/getting-started/data.md

```diff
@@ -0,0 +1,1202 @@
+
+```js
+const store = wp.data.createReduxStore( 'thermostat', {
+	// Essential functions
+	selectors: {
+		getTemperatureCelcius: ( state ) => state.temperature,
+		getTemperatureFarenheit: ( state ) => state.temperature * 1.8 + 32
+	},
+	resolvers: {
+		getTemperatureCelcius: () => ( { dispatch } ) => {
+			dispatch.receiveTemperature( 10 );
+		}
+	},
+
+	// Utility functions
+	__experimentalUseThunks: true,
+	actions: {
+		receiveTemperature: ( temperature ) => ({
+			type: 'RECEIVE_TEMPERATURE',
+			temperature
+		})
+	},
+	reducer(state={}, action) {
+		const newState = {
+			...state
+		}
+		if ( action.type === 'RECEIVE_TEMPERATURE' ) {
+			newState.temperature = action.temperature;
+		} else if ( action.type === '@@INIT' ) {
+			newState.temperature = 0;
+		}
+		return newState;
+	}
+} );
+wp.data.register(store)
+```
+
+The `@@INIT` action is dispatched when the store is instantiated, and so the initial state says `temperature: 0`
+
+### Simple selectors
+The  `getTemperatureFarenheit` is a simple selector, it predictably returns `0` once the store was instantiated:
+
+```js
+> wp.data.select('my-store').getTemperatureFarenheit()
+0
+```
+
+Note we didn’t provide the  `state` as an argument. The Gutenberg data layer handles that for us.
+
+### Memoized selectors
+
+dependency for other selectors
+
+```js
+import createSelector from 'rememo';
+
+// This selector will only calculate the return value once,
+// as long as `state.temperature` remains the same.
+getTemperatureFarenheit: createSelector(
+	// The selector
+	( state ) => state.temperature * 1.8 + 32,
+
+	// The reference(s) upon which the computation depends:
+	( state ) => [ state.temperature ]
+)
+```
+
+Read more about memoized selectors in [rememo](https://github.com/aduth/rememo) package documentation.
+
+### Resolved selectors
+`getTemperatureCelcius` is more special as there is a resolver registered under the same name. When `getTemperatureCelcius` is called for the first time, it will receive the current state and return `0`, but the data layer will also call the related resolver. Since our resolver populates the state with the temperature, the second call will return the actual data:
+```js
+> wp.data.select('thermostat').getTemperatureCelcius()
+0
+> wp.data.select('thermostat').getTemperatureCelcius()
+10
+```
+
+Once the data is loaded, `getTemperatureFarenheit` can do something with it:
+```js
+> wp.data.select('thermostat').getTemperatureFarenheit()
+50
+```
+
+As we’re about to learn, the resolvers may be asynchronous. How do you know when the data becomes available? The easiest way is to use the `resolveSelect` utility instead of `select`:
+```js
+> wp.data .resolveSelect('thermostat') .getTemperatureCelcius() .then(( temperature ) => console.log( temperature ))
+10
+```
+`resolveSelect` returns a promise that waits until the resolver finishes, runs the selector, then yields the final value.
```

Comment:
```suggestion
`resolveSelect` returns a promise that _waits until the resolver finishes_, runs the selector, then yields the final value.
```

Reply:
Do you think the italics are useful here?